lib/segment/src/segment.rs

```diff
@@ -766,6 +806,25 @@ impl SegmentEntry for Segment {
         }
     }
 
+    fn update_vectors(
+        &mut self,
+        op_num: SeqNumberType,
+        point_id: PointIdType,
+        vectors: NamedVectors,
+    ) -> OperationResult<bool> {
+        let internal_id = self.id_tracker.borrow().internal_id(point_id);
+        match internal_id {
+            // Point does already not exist anymore
+            None => Ok(false),
```

Comment:
I think we should 404 here

Reply:
Good point!

This originates from deleting points, where it did make sense.

Reply:
https://github.com/qdrant/qdrant/blob/39d093902216afd015727e941e1c58c462912ea7/lib/segment/src/segment.rs#L817-L819

Done for both `update_vectors` and `delete_vectors`.